src/test/java/org/broadinstitute/hellbender/utils/haplotype/HaplotypeBAMWriterUnitTest.java

```diff
@@ -0,0 +1,247 @@
+package org.broadinstitute.hellbender.utils.haplotype;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.hellbender.utils.GenomeLoc;
+import org.broadinstitute.hellbender.utils.genotyper.*;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.UnvalidatingGenomeLoc;
+import org.testng.Assert;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.*;
+
+
+public class HaplotypeBAMWriterUnitTest extends BaseTest {
```

Comment:
`HaplotypeBAMWriter create(final Type type, final File outputFile, final SAMFileHeader sourceHeader) ` is not tested